pkg/apis/v1alpha5/provisioner.go

```diff
@@ -42,23 +43,29 @@ func (p *Provisioner) Validate(ctx context.Context) (errs *apis.FieldError) {
 }
 
 func (p *Provisioner) SetDefaults(ctx context.Context) {
-	for key, value := range map[string]string{
-		v1alpha5.LabelCapacityType: ec2.DefaultTargetCapacityTypeOnDemand,
-		v1.LabelArchStable:         v1alpha5.ArchitectureAmd64,
-	} {
-		hasLabel := false
-		if _, ok := p.Spec.Labels[key]; ok {
-			hasLabel = true
-		}
-		for _, requirement := range p.Spec.Requirements {
-			if requirement.Key == key {
-				hasLabel = true
-			}
-		}
-		if !hasLabel {
-			p.Spec.Requirements = append(p.Spec.Requirements, v1.NodeSelectorRequirement{
-				Key: key, Operator: v1.NodeSelectorOpIn, Values: []string{value},
-			})
-		}
+	requirements := scheduling.NewNodeSelectorRequirements(p.Spec.Requirements...)
+
+	// default to amd64
+	if !requirements.Has(v1.LabelArchStable) {
+		p.Spec.Requirements = append(p.Spec.Requirements, v1.NodeSelectorRequirement{
+			Key: v1.LabelArchStable, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.ArchitectureAmd64},
+		})
+	}
+
+	// default to on-demand
+	if !requirements.Has(v1alpha5.LabelCapacityType) {
+		p.Spec.Requirements = append(p.Spec.Requirements, v1.NodeSelectorRequirement{
+			Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{ec2.DefaultTargetCapacityTypeOnDemand},
+		})
+	}
+
+	// default to C, M, R categories if no instance type constraints are specified
+	if !requirements.Has(v1.LabelInstanceTypeStable) &&
+		!requirements.Has(v1alpha1.LabelInstanceFamily) &&
+		!requirements.Has(v1alpha1.LabelInstanceCategory) &&
+		!requirements.Has(v1alpha1.LabelInstanceGeneration) {
+		p.Spec.Requirements = append(p.Spec.Requirements, v1.NodeSelectorRequirement{
+			Key: v1alpha1.LabelInstanceCategory, Operator: v1.NodeSelectorOpIn, Values: []string{"c", "m", "r"},
```

Comment:
This doesn't match the default we had before, do you want to put a Generation > 2 as well?